public/components/FeatureAnywhereContextMenu/AnywhereParentFlyout/AnywhereParentFlyout.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import './styles.scss';
+import { get } from 'lodash';
+import AddAnomalyDetector from '../CreateAnomalyDetector';
+
+const AnywhereParentFlyout = ({ startingFlyout, ...props }) => {
```

Comment:
We should make the `props` typesafe here to be consistent with the rest of the plugin, and so it is clear what fields exist in `props`. Given our codebase is mostly all typescript, any javascript-like code copied from alerting should be converted here

Reply:
Jackie can probably add more info or give it a try but I was playing around with typesafing this file and it wasn't too straightforward and I had to do something like what I pasted below (not completely done).

If you want to get it working @jackiehanyang feel free to do so or I can add typesafety in my associated detectors PR since I already have context and it might make more sense as I add an additional flyout type. Also let me know if I am over complicating it or doing it wrong with my intution below @ohltyler but the reason for the difficulty is because we instantiate the AnywhereParentFlyout with different files in the final product like the associated flyout and the existing flyout and create.
```
const Flyout: ({ embeddable, closeFlyout, setMode }: AssociatedDetectorsProps | ExistingDetectorsProps) => JSX.Element = {
    associated: AssociatedDetectors,
    existing: CreateAnomalyDetector,
  }[mode];
```

and also instantiate container like this
```
export const Container = ({ startingFlyout, ...props }: { startingFlyout: keyof FlyoutType, [key: string]: any }) => {
```

Reply:
Got it. Even just having a `AnywhereParentFlyoutProps` with some optional fields (depending on if instantiated with association or existing) I think would be fine here. But having some idea of what `props` is should be visible so it's much easier to read.